code/lib/cli-storybook/src/automigrate/fixes/remove-docs-autodocs.test.ts

```diff
@@ -0,0 +1,339 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+
+import type { JsPackageManager, PackageJson } from 'storybook/internal/common';
+import { loadConfig, writeConfig } from 'storybook/internal/csf-tools';
+import type { StorybookConfigRaw } from 'storybook/internal/types';
+
+import type { CheckOptions, Fix } from '../types';
+import { removeDocsAutodocs } from './remove-docs-autodocs';
+
+// Mock ConfigFile type
+interface MockConfigFile {
+  getFieldValue: (path: string[]) => any;
+  getSafeFieldValue: (path: string[]) => any;
+  setFieldValue: (path: string[], value: any) => void;
+  removeField: (path: string[]) => void;
+  _ast: Record<string, unknown>;
+  _code: string;
+  _exports: Record<string, unknown>;
+  _exportDecls: unknown[];
+}
+
+// Store mock configs by path
+const mockConfigs = new Map<string, MockConfigFile>();
+
+vi.mock('../helpers/mainConfigFile', () => {
+  const updateMainConfig = vi.fn().mockImplementation(({ mainConfigPath }, callback) => {
+    const config = mockConfigs.get(mainConfigPath);
+
+    if (!config) {
+      throw new Error(`No mock config found for ${mainConfigPath}`);
+    }
+    return callback(config);
+  });
+  return { updateMainConfig };
+});
+
+vi.mock('storybook/internal/csf-tools', () => ({
+  loadConfig: vi.fn().mockImplementation((path) => {
+    const config = mockConfigs.get(path);
+    if (!config) {
+      throw new Error(`No mock config found for ${path}`);
+    }
+    return {
+      parse: () => config,
+    };
+  }),
+  writeConfig: vi.fn(),
+}));
+
+const mockPackageManager = {
+  retrievePackageJson: vi.fn().mockResolvedValue({
+    dependencies: {},
+    devDependencies: {},
+  }),
+  runPackageCommand: vi.fn(),
+} as unknown as JsPackageManager;
+
+const mockPackageJson = {
+  dependencies: {},
+  devDependencies: {},
+} as PackageJson;
+
+const baseCheckOptions: CheckOptions = {
+  packageManager: mockPackageManager,
+  mainConfig: {
+    stories: ['../src/**/*.stories.@(js|jsx|ts|tsx)'],
+  } as StorybookConfigRaw,
+  storybookVersion: '9.0.0',
+  configDir: '.storybook',
+};
+
+const typedRemoveDocsAutodocs = removeDocsAutodocs as Required<
+  Fix<{ autodocs: boolean | 'tag' | undefined }>
+>;
+
+describe('remove-docs-autodocs migration', () => {
```

Comment:
The tests are super cryptic, but I understand why you have chosen to do it this way.

I would recommend an alternative approach, though to make it more readable for humans: Instead of mocking `updateMainConfig` I would rather prefer to mock `readFile (node:fs)` and `writeFile (node:fs)`. Then, your assertions will be much simplified, because you can just make a inline snapshot of the transformed configuration and the assertions are actually pretty easy to grasp and it is easy to understand how the config file looked before and after the migration.